File: src/algorithms/two-sum-ii.js
/**
 * Given an array of numbers that is already sorted in increasing order, finds
 * two numbers such that they add up to a specific target number. Returns the
 * indices of the two numbers (the array is 1-indexed, so 1 is added to each
 * index.)
 * 
 * @param {number[]} numbers - An array of numbers sorted in increasing order
 * @param {number} target - A target for two numbers to add up to
 * @returns {number[]} - Returns the indices (1-indexed) of the two numbers
 *     that add up to target
 */
function twoSum (numbers, target) {
  let left = 0;
  let right = numbers.length - 1;
  while (left < right) {
    let sum = numbers[left] + numbers[right];
    if (sum === target) {
      return [left + 1, right + 1];
    } else if (sum > target) {
      right--;
    } else {
      left++;
    }
  }
}

File: src/algorithms/two-sum-ii.py
def twoSum(numbers, target):
    """Given a 1-indexed array of integers numbers that is already sorted in 
    non-decreasing order, find two numbers such that they add up to a specific 
    target number. Let these two numbers be numbers[index1] and numbers[index2] 
    where 1 <= index1 < index2 <= numbers.length.

    Return the indices of the two numbers, index1 and index2, added by one as 
    an integer array [index1, index2] of length 2.

    The tests are generated such that there is exactly one solution. You may 
    not use the same element twice. Your solution must use only constant extra 
    space.

    Args:
        numbers (list[int]): A list of integers.
        target (int): An integer.

    Returns:
        list[int]: Returns a list of integers.
    """
    left = 0
    right = len(numbers) - 1
    while left < right:
        total = numbers[left] + numbers[right]
        if total == target:
            return [left + 1, right + 1]
        elif total > target:
            right -= 1
        else:
            left += 1
    return
